### ISSUE TITLE
Add security by checking for signature.

### ISSUE THREAD

**Issue by mordechaim** [edvin/fxlauncher#82]
Using an automatic launcher opens up the user's computer to malicious files. If -- in any event -- the remote server is compromised a hacker can cause automatic download of their file to all users.

The most basic security could be added by bundling the developer's public certificate in the installation and passing the file location to the launcher as a command-line argument. It should be matched with the jar-signing on every update. It should reject the file if it fails.

Of course this could be completely optional, but it is an extremely important addition if this project is used for a popular software.

**Comment by edvin** [comment]
I've been wanting to do something like this for a while, and it's a must for popular software as you point out. My plate is completely full at the moment, so unless somebody beats me to it it's going to be a while before I can get to it, but this needs to be done for sure :)

**Comment by MouseEvent** [comment]
No problem, I'll fork you. I'm very enthusiastic for this project.

**Comment by edvin** [comment]
Great to hear :) I will of course be available to discuss the feature if you need to bounce some ideas :)

**Comment by mordechaim** [comment]
After much thought I came up with this design:

- First, allow all filetypes to specify os dependency by ``-[os].[ext]``, not just jars. (Btw you had a bug in the regex, it would match ``app-wintjar`` too, since you didn't escape the dot.) This will allow ``.zip``'s too.
- Every archive -- except jars -- should automatically be extracted. If you want an archive in the end result, double-wrap it inside another archive.
- If no ``--cert`` option is passed in launcher, just download and extract archives as usual.
- If the ``--cert`` option is used, every library file **must** be an archive and signed with ``jarsigner``, this will solve signing of non-jar native executables. (This actually forced me to design the auto-extract feature above). Will also solve 3rd party signed jars, since it will be re-signed by the wrapping archive without overwriting existing signature.  We now download them all in a temp folder, check for signature validity and extract all *signed* entries (will ignore ``META-INF/MANIFEST.MF`` and other signature data files) except for jars which will just be moved to the cache-dir.


All beautifully set, but one little problem, the checksum... The extracted files will never have the same checksum as the archived file mentioned in the manifest. The extract feature is vital, otherwise I can only validate jars, but will allow ``.exe``'s without a problem. I still think of a solution for this.

Any comment?

------------------

Never mind my username, I recently changed it :)
